framework/include/meshgenerators/MeshGenerator.h

```diff
@@ -181,6 +184,16 @@ class MeshGenerator : public MooseObject, public MeshMetaDataInterface
    */
   bool hasGenerateData() const { return hasGenerateData(_pars); }
 
+  /**
+   * @return If this generator was created as a subgenerator, the generator that created it
```

Comment:
better to call that one the parent than the child?
there could be a subMG in between the two too so it s not the next MG to run either

Reply:
no, it's not a parent. subgenerators are only ever created as parents, thus below is the child. standard with the rest of the nomenclature with parent-child

Reply:
there's still more than one child if we use this convention that does not work as well now that we have subgenerator

getSubgeneratorCreator fits better

Reply:
moot point, just removed the method it was only needed with the previous methodology